game/src/main/scala/io/aigar/game/EntityContainer.scala

```diff
@@ -0,0 +1,22 @@
+package io.aigar.game
+
+import com.github.jpbetz.subspace.Vector2
+
+
+trait EntityContainer {
+  def respawn(grid: Grid, players: List[Player], respawnRetryAttempts: Int): Option[Vector2] = {
+    var newPosition = grid.randomPosition
+
+    1 to respawnRetryAttempts foreach { _ =>
+      for (player <- players) {
+        for (cell <- player.cells) {
+          if (cell.contains(newPosition))
+            newPosition = grid.randomPosition
+          else
+            return Some(newPosition)
```

Comment:
I think this should be at the end of the method, or else it will always only try once.